docs/public-offer/index.rst

```diff
@@ -0,0 +1,242 @@
+Public Offer
+============
+
+To create a new pubic AMI product listing in `AWS marketplace management portal`_.
+
+
+Create product ID
+-----------------
+
+.. code-block::
+
+   awsmp public-offer create
+
+   ChangeSet created (ID: gxy13m673kmhr4vdtpu0ltwf)
+   https://aws.amazon.com/marketplace/management/requests/gxy13m673kmhr4vdtpu0ltwf
+
+This request will generate an offer id associated with product id. An offer ID is **required**
+to update the instance type, the legal term and to release the listing, as seen in the API calls ahead.
+
+.. note::
+
+       For existing published listing, an offer ID can also be obtained from the AWS console under :guilabel:`AWS Marketplace` > :guilabel:`Manage subscriptions`. Select the required listing and look for the `Offer ID` under `Agreement`.
+
+       If you are in draft, go to your request from running this command and check :guilabel:`Entities` section under :guilabel:`Change request summary`.
+
+
+Add/Edit product description
+----------------------------
+
+Once a product ID is created, you can add/edit the product description. The description fields are shown
+below or you can also refer to the sample config file (listing_configuration.yaml)
+
+.. code-block::
+
+   description:
+      product_title: str
+      logourl: str
+      video_urls: Optional[List[str]], can only have 1 url
+      short_description: str
+      long_description: str
+      highlights: List[str]
+      search_keywords: List[str]
+      categories: List[str]
+      support_description: str # Don't include space character at the beginning/end
+      support_resources: str
+      additional_resources: Optional[List[Dict[str, str]]]
+      sku: Optional[str]
+
+For empty values, use ``~`` for str type and ``[]`` for List type
+
+To add/edit the product description, run:
+
+.. code-block::
+
+   awsmp public-offer update-description \
+      --product-id prod-xwpv7txqxg55e
+      --config listing_configuration.yaml
+
+If a field value does not match the required format, it will show up as an error.
+
+
+Update/Add instance type
+------------------------
+
+To update/add an instance type, you need to generate an instance type file and provide it as input while updating the listing.
+
+#. Generate an instance type file (.csv) and provide as input file when updating listing.
+
+*example instance_type.csv*
+
+.. code-block::
+
+   m7a.8xlarge,0.00,0.00
+   m7a.large,0.00,0.00
+   m7a.medium,0.00,0.00
+   m7a.xlarge,0.00,0.00
+   m7i-flex.8xlarge,0.00,0.00
+   m7i-flex.large,0.00,0.00
+   m7i-flex.xlarge,0.00,0.00
+
+You can generate an instance type file in two ways:
+
+#. Using the public-offer command
+
+If you've created new listing, you can use it's architecture and virtual type to run:
+
+.. code-block::
```

Comment:
recommend adding `sh` as type to avoid arbitrary highlighting. (can also use `console` but it adds italics that I'm not personally a fan of)

```suggestion
.. code-block:: sh
```